### PR TITLE
[Cling] Improve help message for .x, allow macros with main.

### PR DESCRIPTION
**Pull request by hageboeck**
- When trying to execute a macro with .x and the corresponding function is
not found, a more informative warning is issued.
- Further, the interpreter can now fall back to main() if defined.

In local tests, I didn't manage to trigger a clash with the "real" main. I tried to define both
- `int main()` and
- `int main(int argc, char** argv)`,
but it just compiles and does the right thing (either it calls with the correct parameters or it complains that the parameters don't match).

### CODE REVIEW

**Comment by hageboeck** [comment]
Removed the fallback to `main()`. The message now reads:
```
warning: Failed to call `LHCb002(1, (const char**)nullptr)` to execute the macro. Add this function or rename the macro. Falling back to `.L`.
```

**Comment by amadio** [comment]
A new line somewhere to break the message in two lines wouldn't hurt.

**Comment by hageboeck** [comment]
> A new line somewhere to break the message in two lines wouldn't hurt.

Done (by editing on github)! Please squash this away when merging.

File: interpreter/cling/lib/MetaProcessor/MetaSema.cpp [hunk 1]
```diff
@@ -151,13 +151,14 @@ namespace cling {
 
       if (expression.empty()) {
         using namespace clang;
+        const char msg[] = "Failed to call `%0%1` to execute the macro.\n"
```
Comment (Axel-Naumann) [inline interpreter/cling/lib/MetaProcessor/MetaSema.cpp]:
```suggestion
        static const char msg[] = "Failed to call `%0%1` to execute the macro.\n"
```

Comment (hageboeck) [reply interpreter/cling/lib/MetaProcessor/MetaSema.cpp]: Interesting. I tested on godbolt, and the difference is that in the current version, it's packing the string into instructions whereas for the static version it's directly in the object file.
Could you elaborate why you prefer the latter?

Comment (hageboeck) [reply interpreter/cling/lib/MetaProcessor/MetaSema.cpp]: ... because it doesn't go on the stack, probably.

**Pull request merged**